### ISSUE TITLE
The accuracy is not as good as said in the paper

### ISSUE THREAD

**Issue by wmonica** [ifp-uiuc/do-neural-networks-learn-faus-iccvw-2015#1]
Has anyone tried this code before? I run the code, the accuracy is just about 50%, lower than the paper mentioned (around 96%). Could any one tell me why?

**Comment by pkhorrami4** [comment]
Hello. I am sorry the code is not running as it should. I believe I was able to reproduce the problem you are experiencing. I think it has to do with the input images. When the images are resized to 96x96, they are returned as data type float64 instead of uint8 resulting in a range mismatch when they are saved as uint8 to the numpy (.npy) file. 

I think if you display an image from the X.npy file, it will probably look like noise rather than a face. I made a correction to the code and posted it to a branch called [fix_image_range](https://github.com/ifp-uiuc/do-neural-networks-learn-faus-iccvw-2015/tree/fix_image_range). Hopefully, this should fix it.

**Comment by wmonica** [comment]
![image](https://user-images.githubusercontent.com/4161303/28216331-d560652a-68e3-11e7-8c79-1a8444dc16de.png)
I have tried as you said. The accuracy is a little higher than before. But the accuracy is still lower then 55%, and I plot the train/test accuracy figure, as shown. I found the training process is overfitting. Could tell me how to solver the problem? Or how I can get the result just as you said. Thanks!

**Comment by pkhorrami4** [comment]
Could you try and display a few of the images for me? The code should be saving the images to a file named X.npy. Do the images look like faces or like noise?

**Comment by wmonica** [comment]
The X.npy still has some noise. As shown.

**Comment by wmonica** [comment]
![0](https://user-images.githubusercontent.com/4161303/28236010-85118328-694d-11e7-9f7b-6ab48684c2a6.png)
![5](https://user-images.githubusercontent.com/4161303/28236008-851099ae-694d-11e7-8c7f-d06b494b2aac.png)
![6](https://user-images.githubusercontent.com/4161303/28236009-8510ce92-694d-11e7-97d6-d4f0f0bb5249.png)
![7](https://user-images.githubusercontent.com/4161303/28236011-8511fa1a-694d-11e7-951f-b5f0a844918f.png)
![8](https://user-images.githubusercontent.com/4161303/28236012-8546838e-694d-11e7-95c2-6ef460790953.png)
![9](https://user-images.githubusercontent.com/4161303/28236014-854893b8-694d-11e7-9475-b44a1228f421.png)
![10](https://user-images.githubusercontent.com/4161303/28236013-85486848-694d-11e7-8fe9-c8f73f092ef2.png)
![11](https://user-images.githubusercontent.com/4161303/28236015-854930a2-694d-11e7-816c-03f72421bc89.png)
The X.npy still has some noise. As shown.

**Comment by pkhorrami4** [comment]
I received an email in which you showed that the images were just noise (strangely, I don't see your message here). As I mentioned before, I believe the issue is because the images are being saved out to the .npy files incorrectly. 

Just to be clear, did you rerun the [make_ck_dataset.py](https://github.com/ifp-uiuc/do-neural-networks-learn-faus-iccvw-2015/blob/25441e5057611ed1d97cbb78f40f3325c24cf1c1/data_scripts/make_ck_plus_dataset.py) file in the [fix_image_range](https://github.com/ifp-uiuc/do-neural-networks-learn-faus-iccvw-2015/tree/fix_image_range) branch to reconstruct the .npy data files before retraining the model? If not, could you please rerun the script, save the .npy files to a temp directory and confirm that the images look like faces. Thanks

**Comment by wmonica** [comment]
Thank you for your help. I checked the .npy file, and the images are just noise. Then I reconstruct the .npy data, now the images are just like faces. Just as shown below.
![0](https://user-images.githubusercontent.com/4161303/28243920-a04e1e60-6a0d-11e7-8494-da0746128eb8.png)
![1](https://user-images.githubusercontent.com/4161303/28243921-a0548462-6a0d-11e7-9014-eb47d314841e.png)
![2](https://user-images.githubusercontent.com/4161303/28243923-a06a55bc-6a0d-11e7-81e7-f8a90cbe69f1.png)
![3](https://user-images.githubusercontent.com/4161303/28243922-a058d922-6a0d-11e7-9678-9ed53b547332.png)
And the training process is very smoothly. The test accuracy is very high.
![image](https://user-images.githubusercontent.com/4161303/28243913-3cc6b276-6a0d-11e7-8b7c-e9195f46629f.png)
Thank you very much.

**Comment by pkhorrami4** [comment]
No problem. Thanks for bringing the issue to my attention.
